app/src/main/java/campus/tech/kakao/map/RecentSearchAdapter.kt

```diff
@@ -0,0 +1,37 @@
+package campus.tech.kakao.map
+
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.BaseAdapter
+import android.widget.TextView
+
+class RecentSearchAdapter(private val recentDataList: List<String>) : BaseAdapter() {
+    override fun getCount(): Int = recentDataList.size
+
+    override fun getItem(position: Int): Any = recentDataList[position]
+
+    override fun getItemId(position: Int): Long = position.toLong()
+
+    override fun getView(position: Int, convertView: View?, parent: ViewGroup?): View {
+        val view: View
+        val recentDataViewHolder : RecentViewHolder
+        if (convertView == null){
```

Comment:
```suggestion
   val view =     if (convertView == null){
```
위와같이 코드를 작성하면 불필요한 선언 코드를 제거할 수 있습니다.

Reply:
```
override fun getView(position: Int, convertView: View?, parent: ViewGroup?): View {
        val view: View
        val recentDataViewHolder : RecentViewHolder
        if (convertView == null){
            view = LayoutInflater.from(parent?.context).inflate(R.layout.recent_search_item,parent,false)
            recentDataViewHolder = RecentViewHolder(view)
            view.tag = recentDataViewHolder
        } else{
            view = convertView
            recentDataViewHolder = convertView.tag as RecentViewHolder
        }
        val item = recentDataList[position]
        recentDataViewHolder.name.text = item

        return view
    }
```
if문에서 ViewHolder와 tag 설정을 같이 하기 때문에 view에 바로 값을 넣는 형식은 적용이 어려울 것 같습니다. 확인부탁드립니다!

Reply:
```suggestion
       val view = convertView ?: LayoutInflater.from(parent?.context).inflate(R.layout.recent_search_item,parent,false)
        val recentDataViewHolder = view.tag as? RecentViewHolder ?: RecentViewHolder(view).also { view.tag = it }
        recentDataViewHolder.name.text = getItem(position) as String
```
이렇게 간단하게 작성해볼 수 있을 것 같아요~!

Reply:
우와...! 정말 멋있습니다...!
코틀린의 매력은 엄청난 것 같습니다

코드 수정했습니다! 피드백 감사합니다!!!